sdk/security_keyvault/src/clients/pipeline.rs

```diff
@@ -0,0 +1,32 @@
+use crate::clients::policy::{AuthorizationPolicy, TimeoutPolicy};
+use azure_core::{auth::TokenCredential, ClientOptions, Pipeline};
+use std::sync::Arc;
+
+pub(crate) fn new_pipeline_from_options(
+    credentials: Arc<dyn TokenCredential>,
+    scope: String,
+) -> Pipeline {
+    let auth_policy: Arc<dyn azure_core::Policy> =
+        Arc::new(AuthorizationPolicy::new(credentials, scope));
```

Comment:
In cosmos we're sending the equivalent of `scope` as part of the `Context`. Would that perhaps work better here too?